pkg/cloud/vsphere/context/cluster_context.go

```diff
@@ -344,7 +341,10 @@ func (c *ClusterContext) Patch() error {
 		c.Cluster.ResourceVersion = result.ResourceVersion
```

Comment:
shouldn't we simply do
`c.Cluster = result`
That way `c.Cluster` always points to the latest persisted version?

Reply:
Hi @sidharthsurana,

Patching is the last thing that happens, so the RV only matters for the status update really. And besides, the RV *is* updated on the the existing cluster. It's also what CAPA does FWIW.

Reply:
True, but my only concern is that are we making an assumption that only this code is going to be the one that will ever mutate the Cluster or Machine object?
In the future, there may be more controller looking and possibly mutating the same object. Thus I feel in a probably very slim chance but we may overwrite someone else's changes done via outside this controller.

Reply:
No, this will always be the last code per the defer. If it's not we can revisit. Either way, the resource version *is* updated. :) I just don't want to make that change here.

Reply:
That's right, if the patch fails, then we return an error. Otherwise, if the patch succeeds, then the cluster object *is* the same as the one on the server.

Reply:
I guess for now it is okay since there are no other additional controllers mutating these objects.
If I think about it again, the risk here is probably not that this method doesn't successfully update the server object. But rather it is that after this update the `c.Cluster` object may not capture the most current persisted object.